src/System.Private.CoreLib/shared/System/Globalization/StringInfo.cs

```diff
@@ -15,18 +16,18 @@ public class StringInfo
     {
         private string _str;
 
-        private int[] _indexes;
+        private int[]? _indexes;
 
         public StringInfo() : this(string.Empty)
         {
         }
 
         public StringInfo(string value)
         {
-            this.String = value;
+            _str = value;
```

Comment:
This is changing behavior.  Previously if null was passed in we would throw an ArgumentNullException; now it doesn't.  Please fix that.

Reply:
but now it won't compile if someone passes null, right? will add a check.
I changed this logic because the compiler complained that _str is unitialized and marked as nullable - it couldn't see that property sets the value so it complained it as an error. cc: @jaredpar

Reply:
> but now it won't compile if someone passes null, right?

Only if someone turns on the nullability feature and doesn't suppress the warning.  We need to not change behavior when doing this annotation.

Reply:
I've reverted the logic here but had to add a hack for compiler complaints (`_str = _str!;`)